### PR TITLE
#45 add unit test for TkFallback

### PR DESCRIPTION
**Pull request by dmzaytsev**

### CODE REVIEW

**Comment by davvd** [comment]
Thanks, someone will review your pull request soon

**Comment by davvd** [comment]
@longtimeago this pull request is for you, please review

File: src/test/java/org/takes/facets/fallback/TkFallbackTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ */
+public final class TkFallbackTest {
+    /**
+     * TkFallback can fall back.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void fallsBack() throws IOException {
+        final String err = "message";
+        final String format = "something went wrong: %s";
```
Comment (longtimeago) [inline src/test/java/org/takes/facets/fallback/TkFallbackTest.java]: format is a noise, let's use simple message make an assertion on it

File: src/test/java/org/takes/facets/fallback/TkFallbackTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        MatcherAssert.assertThat(
+            new RsPrint(
+                new TkFallback(
+                    new Take() {
+                        @Override
+                        public Response act() {
+                            throw new UnsupportedOperationException(err);
```
Comment (longtimeago) [inline src/test/java/org/takes/facets/fallback/TkFallbackTest.java]: what if ```Take``` throws ```Throwable``` ?

**Comment by longtimeago** [comment]
@dmzaytsev Thanks! See several comments above in tha code

File: src/test/java/org/takes/facets/fallback/TkFallbackTest.java (hunk 2 above)
Comment (dmzaytsev) [inline src/test/java/org/takes/facets/fallback/TkFallbackTest.java]: I think Take can't throws Throwable in this case.  Throwable is checked exception, which checked by compiler.

Comment (longtimeago) [inline src/test/java/org/takes/facets/fallback/TkFallbackTest.java]: ah yep, right

Comment (dmzaytsev) [inline src/test/java/org/takes/facets/fallback/TkFallbackTest.java]: Could be replaced by Runtime Exception, but PMD restricts throwing raw exception types

**Comment by dmzaytsev** [comment]
@longtimeago review again please

File: src/test/java/org/takes/facets/fallback/TkFallbackTest.java (hunk 2 above)
Comment (longtimeago) [inline src/test/java/org/takes/facets/fallback/TkFallbackTest.java]: I thinks this is the time to shut up PMD. We need to make sure we intercept all possible runtime exceptions

Comment (dmzaytsev) [inline src/test/java/org/takes/facets/fallback/TkFallbackTest.java]: ok, I did. Review again please

**Comment by longtimeago** [comment]
@dmzaytsev ok, thanks!

**Comment by longtimeago** [comment]
@rultor merge please

File: src/test/java/org/takes/facets/fallback/TkFallbackTest.java (hunk 2 above)
Comment (dmzaytsev) [inline src/test/java/org/takes/facets/fallback/TkFallbackTest.java]: @longtimeago I changed to IOException as  in Take.act's declaration
Take implementations could throws IOException descendants and unchecked exceptions

**Comment by yegor256** [comment]
@rultor merge it

**Comment by davvd** [comment]
@longtimeago **18 mins** was added to your account, many thanks for your contribution! The task took 54223466.... review comments (c=3) added as a bonus... added +18 to your rating, now it is equal to [+3693](http://www.netbout.com/b/35024?open=rating)

**Comment by davvd** [comment]
@rultor deploy

**Pull request merged**